sklearn/metrics/classification.py

```diff
@@ -1249,6 +1305,17 @@ class 2       1.00      0.67      0.80         3
                                                   average=None,
                                                   sample_weight=sample_weight)
 
+    # compute average metrics for all methods in averages
+    average_metrics = (precision_recall_fscore_support(y_true, y_pred,
+                                                       labels=labels,
+                                                       average=el,
+                                                       sample_weight=
+                                                       sample_weight)
+                       # skip calculating average_metrics in case of a binary
+                       # classification
+                       if el else None
+                       for el in averages)
```

Comment:
I think we should special case binary classification as the underlying `precision_recall_fscore_support` does it: f1, precision and recall only make sense when one class is treated as the "positive" class.

Furthermore there is no point in averaging the precision of the positive class with the precision of the negative class in my opinion. We can just display the precision / recall / f1 for each class treated has positive and let the user choose which line to read.